vortex-datafusion/src/persistent/opener.rs

```diff
@@ -61,130 +69,58 @@ pub(crate) struct VortexOpener {
-                phys_field.is_nullable(),
-            )))
-        }
-        _ => table_field.data_type().clone(),
-    }
-}
-
-/// Merges two field collections, using logical types from table_fields where available.
-/// Falls back to physical field types when no matching table field is found.
-fn merge_fields(
-    physical_fields: &arrow_schema::Fields,
-    table_fields: &arrow_schema::Fields,
-) -> Vec<Field> {
-    physical_fields
-        .iter()
-        .map(|phys_field| {
-            table_fields
-                .iter()
-                .find(|f| f.name() == phys_field.name())
-                .map(|table_field| {
-                    Field::new(
-                        phys_field.name(),
-                        merge_field_types(phys_field, table_field),
-                        phys_field.is_nullable(),
-                    )
-                })
-                .unwrap_or_else(|| (**phys_field).clone())
-        })
-        .collect()
-}
-
-/// Computes a logical file schema from the physical file schema and the table
-/// schema.
-///
-/// For each field in the physical file schema, looks up the corresponding field
-/// in the table schema and uses its logical type.
-fn compute_logical_file_schema(
-    physical_file_schema: &SchemaRef,
-    table_schema: &SchemaRef,
-) -> SchemaRef {
-    let logical_fields: Vec<Field> = physical_file_schema
-        .fields()
-        .iter()
-        .map(|physical_field| {
-            table_schema
-                .fields()
-                .find(physical_field.name())
-                .map(|(_, table_field)| {
-                    Field::new(
-                        physical_field.name(),
-                        merge_field_types(physical_field, table_field),
-                        physical_field.is_nullable(),
-                    )
-                    .with_metadata(physical_field.metadata().clone())
-                })
-                .unwrap_or_else(|| (**physical_field).clone())
-        })
-        .collect();
-
-    Arc::new(arrow_schema::Schema::new(logical_fields))
-}
-
 impl FileOpener for VortexOpener {
     fn open(&self, file_meta: FileMeta, file: PartitionedFile) -> DFResult<FileOpenFuture> {
         let session = self.session.clone();
         let object_store = self.object_store.clone();
         let projection = self.projection.clone();
         let mut filter = self.filter.clone();
         let file_pruning_predicate = self.file_pruning_predicate.clone();
-        let expr_adapter_factory = self.expr_adapter_factory.clone();
+
+        // We create our custom expression and schema adapters. The schema adapter is just the
+        // DF default adapter. Our expression adapter is currently built to work around a bug
+        // in the upstream adapter.
+        // See: https://github.com/apache/datafusion/issues/17114
+        let expr_adapter_factory = DEFAULT_EXPR_ADAPTER.clone();
+        let schema_adapter_factory = DEFAULT_SCHEMA_ADAPTER.clone();
```

Comment:
Should we return an error in https://github.com/vortex-data/vortex/blob/2f80934efa7e66595dafa365d6e32cf4e0b1d395/vortex-datafusion/src/persistent/source.rs#L304-L307

on the source? It might be confusing to use that and then figure out why the provided schema adapter factory is not being used. Or maybe in a more backwards-compatible way use the provided schema adapter factory but log a warning.